layout.go

```diff
@@ -243,6 +263,183 @@ func (r *recordReader) guid() (v [16]byte) {
+	if kind > uint8(flags.SigKind_VARARG) {
+		r.err = fmt.Errorf("signature kind is not a method def signature: %v", kind)
+		return
+	}
+	thisiness := firstByte & 0xF0
+	v.HasThis = thisiness&uint8(flags.SigAttributes_HASTHIS) != 0
+	v.ExplicitThis = thisiness&uint8(flags.SigAttributes_EXPLICITTHIS) != 0
+	if thisiness&uint8(flags.SigAttributes_GENERIC) != 0 {
+		v.Generic = sigR.compressedUint32()
+		if fail() {
+			return
+		}
+	}
+	paramCount := sigR.compressedUint32()
+	if fail() {
+		return
+	}
+
+	v.RetType = sigR.retType()
+	if fail() {
+		return
+	}
+	for i := uint32(0); i < paramCount; i++ {
+		v.Param = append(v.Param, sigR.param())
+		if fail() {
+			return
+		}
+	}
+	return
+}
+
+func (r *recordReader) param() (v SigParam) {
+	if r.err != nil {
+		return
+	}
+	v.Type = r.decodeType()
+	switch v.Type.Kind {
+	case flags.ElementType_BYREF:
+		v.Kind = ParamKind_ByRef
+	case flags.ElementType_TYPEDBYREF:
+		v.Kind = ParamKind_TypedByRef
+	default:
+		v.Kind = ParamKind_ByValue
+	}
+	return
+}
+
+func (r *recordReader) retType() (v RetType) {
+	if r.err != nil {
+		return
+	}
+	v.Type = r.decodeType()
+	switch v.Type.Kind {
+	case flags.ElementType_BYREF:
+		v.Kind = RetTypeKind_ByRef
+	case flags.ElementType_TYPEDBYREF:
+		v.Kind = RetTypeKind_ByRef
+	case flags.ElementType_VOID:
+		v.Kind = RetTypeKind_Void
+	default:
+		v.Kind = RetTypeKind_ByValue
+	}
+	return
+}
+
+func (r *recordReader) customModOpt() (v CustomMod) {
+	v.Kind = CustomModKind_Opt
+	v.Index = r.typeHandle()
+	return
+}
+
+func (r *recordReader) customModReqd() (v CustomMod) {
+	v.Kind = CustomModKind_Reqd
+	v.Index = r.typeHandle()
+	return
+}
+
+// typeHandle reads a type handle (a TypeDefOrRefOrSpecEncoded).
+func (r *recordReader) typeHandle() (v CodedIndex) {
+	if r.err != nil {
+		return
+	}
+	value := r.compressedUint32()
+	if r.err != nil {
+		return
+	}
+	// Once we decompress the uint32, we could reverse the encoding steps listed in §II.23.2.8, but
+	// the coded index algorithm has the same result if we define codedTypeDefOrRefOrSpec.
+	v, r.err = parseCoded(codedTypeDefOrRefOrSpec, value)
+	return
+}
+
+func (r *recordReader) decodeType() (v Type) {
+	if r.err != nil {
+		return
+	}
+	switch b := flags.ElementType(r.compressedUint32()); b {
+	// Use recursion to collect the full list of mods, like the System.Reflection.Metadata impl.
+	case flags.ElementType_CMOD_OPT:
+		mod := r.customModOpt()
+		v = r.decodeType()
+		v.Mod = append(v.Mod, mod)
+	case flags.ElementType_CMOD_REQD:
+		mod := r.customModReqd()
+		v = r.decodeType()
+		v.Mod = append(v.Mod, mod)
+
+	case flags.ElementType_BYREF:
+		v.Kind = b
+		v.Value = r.decodeType()
+
+	// TypedByRef and Void have no Type afterwards.
+	case flags.ElementType_TYPEDBYREF:
+		v.Kind = flags.ElementType_TYPEDBYREF
+	case flags.ElementType_VOID:
+		v.Kind = flags.ElementType_VOID
+
+	//case flags.ElementType_GENERICINST:
+	//	v.Kind |= flags.ElementType_GENERICINST
+	//	v.Value = r.genericInst()
+
+	case
```

Comment:
`case` followed by a line jump is not idiomatic, first time a see one.

Reply:
Unfortunate, I like the way it aligns. 😄 I did find some of these in Go (6 cases, e.g. [src/go/parser/parser.go#L1868-L1872](https://github.com/golang/go/blob/15e705ea963b5008112793507365e24b743606bc/src/go/parser/parser.go#L1868-L1872)) but yeah, the overwhelming number of multiline cases in Go don't have a newline here.